pkg/storage/s3/client_test.go

```diff
@@ -1,18 +1,68 @@
 package s3
 
 import (
+	"reflect"
 	"testing"
 
+	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/client-go/kubernetes/scheme"
 	k8sClient "sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/client/fake"
 
 	velerov1alpha2 "github.com/openshift/managed-velero-operator/pkg/apis/managed/v1alpha2"
+	"github.com/openshift/managed-velero-operator/version"
 )
 
+func TestNewS3Client(t *testing.T) {
+	instance := setUpInstance(t)
+
+	tests := []struct {
+		name        string
+		kubeClient  k8sClient.Client
+		region      string
+		expectError bool
+	}{
+		{
+			name:        "create an S3 client",
+			kubeClient:  setUpTestClient(t, instance),
+			region:      "us-east-1",
+			expectError: false,
+		},
+		{
+			name:        "errors without S3 credentials secret",
+			kubeClient:  setUpUnauthedTestClient(t, instance),
+			region:      "us-east-1",
+			expectError: true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			actualClient, err := NewS3Client(tt.kubeClient, tt.region)
+			if (err != nil) && (tt.expectError == false) {
```

Comment:
stylistically, when comparing against a bool, do the following:
`if something == true` -> `if something`
`if something == false` -> `if !something`